### ISSUE TITLE
Collision with Category Names

### ISSUE THREAD

**Issue by jerelevine** [iabudiab/HTMLKit#35]
The NSString+HTMLKit.h and NSCharacterSet+HTMLKit.h categories contain generic named category methods. They are causing collisions with other methods I've already been using. It would be great if you could prefix the methods to reduce potential collisions. Perhaps the prefix could be "html_" or something similar. Thanks!

**Comment by iabudiab** [comment]
@jerelevine Hey there, do you mean all the category methods or just the generic ones? 

Looking at the `NSString` Category, I could hide the first three methods completely, since these are too general. The other three could be prefixed.

```objc
@interface NSString (HTMLKit)
// Hide
- (BOOL)isEqualToStringIgnoringCase:(NSString *)aString;
- (BOOL)isEqualToAny:(NSString *)first, ... NS_REQUIRES_NIL_TERMINATION;
- (BOOL)hasPrefixIgnoringCase:(NSString *)aString;

// Prefix
- (BOOL)isHTMLWhitespaceString;
- (NSUInteger)leadingHTMLWhitespaceLength;
- (NSString *)stringByEscapingForHTML;
@end
```

The `NSCharacterSet` category could also be either completely hidden or prefixed.

```objc
@interface NSCharacterSet (HTMLKit)
+ (instancetype)HTMLWhitespaceCharacterSet;
+ (instancetype)HTMLHexNumberCharacterSet;
+ (instancetype)CSSNthExpressionCharacterSet;
@end
```

**Comment by jerelevine** [comment]
Thanks for responding! Yeah I think like your suggestion. Seems like a perfect solution to me. I don't use the NSCharacterSet category methods but I could see others using them. Probably worth keeping them visible but prefixed.

**Comment by iabudiab** [comment]
Released in [Version 3.0.0](https://github.com/iabudiab/HTMLKit/releases/tag/3.0.0)
